### PR TITLE
Update Parent pom and target sdk deployment jobs for R4_31_maintenance branch

### PR DESCRIPTION
**Pull request by elsazac**
Fixes: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/1847

### CODE REVIEW

File: JenkinsJobs/Releng/deployPlatformParentPom.groovy [hunk 1]
```diff
@@ -31,7 +31,8 @@ For this to be used by consuming projects, they must have their local maven repo
       branch('R4_27_maintenance')
```
Comment (akurtakov) [inline JenkinsJobs/Releng/deployPlatformParentPom.groovy:31]: Please remove deployment of old versions. That should create less churn for old releases.

File: JenkinsJobs/Releng/deploySdkPom.groovy [hunk 2]
```diff
@@ -32,6 +32,7 @@ For this to be used by consuming projects, they must have their local maven repo
       branch('R4_28_maintenance')
```
Comment (akurtakov) [inline JenkinsJobs/Releng/deploySdkPom.groovy:32]: Same

**Review by akurtakov — COMMENTED** [review]
Deploying only master and 4.31 is probably what needed here.

File: JenkinsJobs/Releng/deployPlatformParentPom.groovy (hunk 1 above)
Comment (elsazac) [reply JenkinsJobs/Releng/deployPlatformParentPom.groovy:31]: @akurtakov Thanks for the review. Just to confirm, can remove all the old branches from R4_7_maintenance up till R4_30_maintenance right?

Comment (akurtakov) [reply JenkinsJobs/Releng/deployPlatformParentPom.groovy:31]: I believe it's best. These branches are done and changes to parent poms risk to break builds entirely. Furthermore, there are no verification builds for these branches so deploying untested content is risky.

Comment (elsazac) [reply JenkinsJobs/Releng/deployPlatformParentPom.groovy:31]: Sure. I will remove the same.

**Review by akurtakov — APPROVED** [review]

**Pull request merged**